cmake/modules/CheckYasm.cmake

```diff
@@ -37,7 +37,7 @@ macro(check_yasm_support _object_format _good_result _better_result)
     message(STATUS "Found Yasm: but x86_64 with x32 ABI is not supported")
   elseif(${_better_result})
     message(STATUS "Found Yasm: good -- capable of assembling x86_64")
-  elseif(${_good_result})
-    message(STATUS "Found Yasm: better -- capable of assembling AVX2")
+  elseif(${_support_x64})
+    message(STATUS "Found Yasm: better -- capable of assembling x86_64")
```

Comment:
i don't understand why you are changing the message. on the contrary, i think we should change the message at line 39. because, if we have  `_good _result` implies the amd64 ABI. and `_better_result` implies the capability of assembling AVX2 assembly.

Reply:
@tchaikov 
The original code shows that
```
 36   elseif(${_better_result})
 37     message(STATUS "Found Yasm: good -- capable of assembling x86_64")
 38   elseif(${_good_result})
 39     message(STATUS "Found Yasm: better -- capable of assembling AVX2")
```
I want to change it to be likely in this PR. What's your suggestion?
```
 36   elseif(${_better_result})
 37     message(STATUS "Found Yasm: good -- capable of assembling x86_64 and avx2")
 38   elseif(${_good_result})
 39     message(STATUS "Found Yasm: better -- capable of assembling x86_64")
```

Reply:
i think i've explained my suggestion in previous comment.

Reply:
@tchaikov 
```
if we have _good _result implies the amd64 ABI. and _better_result implies the capability of assembling AVX2 assembly.
```
This is what I have changed in this PR besides changing "_good_result" to "_support_x64". I could keep "_good_result" there, but is it possible to refine the message info?